lib/engine/game/g_1822_pnw/step/merge.rb

```diff
@@ -0,0 +1,214 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../../token'
+require_relative '../../../step/token_merger'
+require_relative '../../../step/programmer_merger_pass'
+
+module Engine
+  module Game
+    module G1822PNW
+      module Step
+        class Merge < Engine::Step::Base
+          include Engine::Step::TokenMerger
+          include Engine::Step::ProgrammerMergerPass
+
+          def actions(entity)
+            return [] if !entity.corporation? || entity != current_entity || @merge_state == :done
+            return ['choose'] if @merge_state != :none
+            return ['merge'] if mergeable(entity).any?
+
+            []
+          end
+
+          def setup
+            @merge_state = :none
+          end
+
+          def merge_name(_entity = nil)
+            'Merge'
+          end
+
+          def description
+            'Merge Corporation'
+          end
+
+          def mergeable(corporation)
+            @game.unassociated_minors.select do |m|
+              @game.graph.connected_nodes(corporation)[m.tokens.first.city] && !valid_par_prices(corporation, m).empty?
+            end
+          end
+
+          def choice_name
+            case @merge_state
+            when :selecting_par
+              'Choose par value for new company'
+            when :selecting_shares
+              'Choose number of shares'
+            when :selecting_token
+              'What to do with the token'
+            end
+          end
+
+          def min_exchange_shares(par, minors_value, minors_cash)
+            ((minors_value - minors_cash - 1) / par).floor + 1
+          end
+
+          def max_exchange_shares(par, minors_value, player_cash)
+            ((minors_value + [player_cash, (par - 1)].min) / par).floor
+          end
+
+          def possible_exchanged_shares(par, minors_cash, minors_value, player_cash)
+            (2..10).to_a.filter do |n|
+              n >= min_exchange_shares(par, minors_value,
+                                       minors_cash) && n <= max_exchange_shares(par, minors_value,
+                                                                                player_cash) && (6 * par) >= minors_value
```

Comment:
min and max don't change. Calculate before the block?

Reply:
Also, if (6 * par) >= minors_value if false, will the block always return false?

Reply:
Yes, they can be calculated before the block.  I'll change it.

It is the intent to always return false in that case - it would be better if that was a check up front.  I'll change it.